molecule/default/verify.yml

```diff
@@ -0,0 +1,15 @@
+---
+- name: Verify user creation
+  hosts: all
+  become: yes
+
+  tasks:
+    - name: Check if user is created
+      stat:
```

Comment:
Stat never fails, even if the path you're testing is absent, so it's not an efficacious test by itself.  You also need to run an assert task on the registered variable(s).